source/Plugins/Language/Swift/SwiftFormatters.cpp

```diff
@@ -180,67 +98,109 @@ bool lldb_private::formatters::swift::StringGuts_SummaryProvider(
+  // platform-independent 128-bit raw value representation from
+  // _StringObject, then interpreting that.
   Status error;
-  static ConstString g_object("_object");
-  static ConstString g_otherBits("_otherBits");
-  static ConstString g_value("_value");
+  uint64_t raw0;
+  uint64_t raw1;
+
+  if (ptrSize == 8) {
+    // On 64-bit platforms, we simply need to get the raw integer
+    // values of the two stored properties.
+    static ConstString g__countAndFlags("_countAndFlags");
+
+    auto countAndFlags =
+      object_sp->GetChildAtNamePath({g__countAndFlags, g__storage, g__value});
+    if (!countAndFlags)
+      return false;
+    raw0 = countAndFlags->GetValueAsUnsigned(0);
+
+    auto object = object_sp->GetChildMemberWithName(g__object, true);
+    if (!object)
+      return false;
+    raw1 = object->GetValueAsUnsigned(0);
+
+  } else if (ptrSize == 4) {
+    // On 32-bit platforms, we emulate what `_StringObject.rawBits`
+    // does. It involves inspecting the variant and rearranging bits
+    // to match the 64-bit representation.
+    static ConstString g__count("_count");
+    static ConstString g__variant("_variant");
+    static ConstString g__discriminator("_discriminator");
+    static ConstString g__flags("_flags");
+
+    auto count_sp = object_sp->GetChildAtNamePath({g__count, g__value});
+    if (!count_sp) return false;
+    uint64_t count = count_sp->GetValueAsUnsigned(0);
+
+    auto discriminator_sp =
+      object_sp->GetChildMemberWithName(g__discriminator, true);
+    if (!discriminator_sp) return false;
+    uint64_t discriminator = discriminator_sp->GetValueAsUnsigned(0);
+    if (discriminator > 0x7F) {
+      // The discriminator only has 7 bits on 32-bit platforms.
+      return false;
+    }
+
+    auto flags_sp = object_sp->GetChildAtNamePath({g__flags, g__value});
+    if (!flags_sp) return false;
+    uint64_t flags = flags_sp->GetValueAsUnsigned(0);
+    if (flags > 0xFFFF) return false;
+
+    auto variant_sp = object_sp->GetChildMemberWithName(g__variant, true);
+    if (!variant_sp) return false;
+
+    llvm::StringRef variantCase = variant_sp->GetValueAsCString();
+
+    ValueObjectSP payload_sp;
+    if (variantCase.startswith("immortal")) {
+      static ConstString g_immortal("immortal");
+      // Set the immortal bit in the discriminator.
+      discriminator |= 0x80;
+      payload_sp = variant_sp->GetChildAtNamePath({g_immortal, g__value});
+    } else if (variantCase.startswith("native")) {
+      static ConstString g_native("native");
+      payload_sp = variant_sp->GetChildMemberWithName(g_native, true);
+    } else if (variantCase.startswith("bridged")) {
+      static ConstString g_bridged("bridged");
+      auto anyobject_sp = variant_sp->GetChildMemberWithName(g_bridged, true);
+      if (!anyobject_sp) return false;
+      payload_sp = anyobject_sp->GetChildAtIndex(0, true); // "instance"
+    } else {
+      return false; // Unknown variant
```

Comment:
Can we make this an `lldbassert` to make sure we catch new variants added in the future?

Reply:
Would this look like?:
```c++
    } else {
      lldbassert("Uknown variant")
      return false;
    }
```